static/app/views/alerts/list/rules/teamFilter.tsx

```diff
@@ -1,119 +1,119 @@
-import {useState} from 'react';
+import {Fragment, useMemo} from 'react';
 import styled from '@emotion/styled';
 import debounce from 'lodash/debounce';
 
-import Input from 'sentry/components/forms/controls/input';
-import LoadingIndicator from 'sentry/components/loadingIndicator';
+import TeamAvatar from 'sentry/components/avatar/teamAvatar';
+import Badge from 'sentry/components/badge';
+import CompactSelect from 'sentry/components/forms/compactSelect';
 import {DEFAULT_DEBOUNCE_DURATION} from 'sentry/constants';
+import {IconUser} from 'sentry/icons';
 import {t} from 'sentry/locale';
-import space from 'sentry/styles/space';
-import {isActiveSuperuser} from 'sentry/utils/isActiveSuperuser';
 import useTeams from 'sentry/utils/useTeams';
 
-import Filter from './filter';
-
 interface Props {
-  handleChangeFilter: (activeFilters: Set<string>) => void;
-  selectedTeams: Set<string>;
-  fullWidth?: boolean;
+  handleChangeFilter: (activeFilters: string[]) => void;
+  selectedTeams: string[];
   /**
    * only show teams user is a member of
    */
   showIsMemberTeams?: boolean;
-  /**
-   * show My Teams and Unassigned options
-   */
-  showMyTeamsAndUnassigned?: boolean;
   /**
    * show My Teams as the default dropdown description
    */
   showMyTeamsDescription?: boolean;
+  /**
+   * show suggested options (My Teams and Unassigned)
+   */
+  showSuggestedOptions?: boolean;
 }
 
+const suggestedOptions = [
+  {
+    label: t('My Teams'),
+    value: 'myteams',
+  },
+  {
+    label: t('Unassigned'),
+    value: 'unassigned',
+  },
+];
+
 function TeamFilter({
   selectedTeams,
   handleChangeFilter,
-  fullWidth = false,
   showIsMemberTeams = false,
-  showMyTeamsAndUnassigned = true,
+  showSuggestedOptions = true,
   showMyTeamsDescription = false,
 }: Props) {
-  const {teams, onSearch, fetching} = useTeams();
-  const debouncedSearch = debounce(onSearch, DEFAULT_DEBOUNCE_DURATION);
-  const [teamFilterSearch, setTeamFilterSearch] = useState<string | undefined>();
-  const isSuperuser = isActiveSuperuser();
+  const {teams, onSearch, fetching} = useTeams({provideUserTeams: showIsMemberTeams});
 
-  const additionalOptions = [
-    {
-      label: t('My Teams'),
-      value: 'myteams',
-      checked: selectedTeams.has('myteams'),
-      filtered: false,
-    },
-    {
-      label: t('Unassigned'),
-      value: 'unassigned',
-      checked: selectedTeams.has('unassigned'),
-      filtered: false,
-    },
-  ];
-  const isMemberTeams = teams.filter(team => team.isMember);
-  const teamItems = (isSuperuser ? teams : showIsMemberTeams ? isMemberTeams : teams).map(
```

Comment:
`useTeams` already checks if the user is a superuser (if yes, then it will always return the full teams list regardless of whether `showIsMemberTeams` is true or not): https://github.com/getsentry/sentry/blob/vl/alert-team-filter/static/app/utils/useTeams.tsx#L340-L342